genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/JobSetupServiceImpl.java

```diff
@@ -530,7 +530,28 @@ String composeScript() {
 
             // Script Environment Section
 
-            this.environmentVariables.forEach(
+            sb.append("# Locally-generated environment variables").append(NEWLINE);
+
+            sb.append(NEWLINE);
+
+            this.localEnvironmentVariables.forEach(
```

Comment:
You might want to move the above `sb.append(NEWLINE)` into the loop as the first step and remove the trailing `NEWLINE` from the loop. That will make it so the new lines are balanced. As it is now after the last ENV variable there will be 2 blank lines. Obviously this is a tiny detail just pointing it out.

Reply:
The double linebreak there was intentional for readability (of the script) since it's between 2 sections of the script.